_docs/rfc/scaffolding_new_modules.md

```diff
@@ -0,0 +1,402 @@
+default, which can be overridden by either forking our `infrastructure-templates` repository or by building new ones
+from scratch.
+
+`terragrunt new` should support the following scenarios:
+
+- `terragrunt new component`, which will add a new component to an existing environment. The inputs for this template
+  should be:
+    - Environment: which live environment folder should the component be added to?
+    - Component module: which module should be used for the component?
+
+- `terragrunt new environment`, which will add a new environment to an existing region. The inputs for this template
+  should be:
+    - Location: which live region/account folder should the environment be added to?
+    - Stack: which stack should be used? This assumes you have different kinds of stacks in your environment, where a
+      stack is a set of components.
+
+- `terragrunt new region`, which will add a new region to an existing account. The inputs for this template should be:
+    - Account: which live account folder should the region be added to? (NOTE: `account` may not be the right
+      terminology here, as it does not mesh with GCP where you have Projects).
+
+- `terragrunt new account`, which will add a new account. The inputs for this template should be:
+    - Remote state info: the minimum information necessary to setup the remote state config.
+
+- `terragrunt new wrapper-module`, which will add a new wrapper module (what you have in `infrastructure-modules`). Note
+  that as such, this will be rendered into `infrastructure-modules`, as opposed to `infrastructure-live` like the other
+  sub commands. The inputs for this template should be:
+    - Module name: what to call this module.
+    - Library Module source: what the underlying library module you are basing the wrapper on.
+
+- `terragrunt new unicorn` (__PLACEHOLDER: I don't know what to call this__. Ideas: `other`, `arbitrary`, `config`),
+  which will support adding any arbitrary template. The inputs depend on the selected template, but at the very least
+  includes:
+    - Template source: where does the template to use live?
+
+### Setup
+
+`terragrunt new` depends on a repository of templates. These templates can practically live anywhere, but for the
+purposes of versioning, it should live outside of `infrastructure-live` and `infrastructure-modules`. For the purposes
+of this example, we will assume all the templates live in a repository called `infrastructure-templates`.
+
+To support each of the scenario above, `terragrunt` will need to be configured to know which subtemplates should be used
+for each of the scenarios, in addition to knowing where the `infrastructure-live` and `infrastructure-modules`
+repositories live. This should be configured in a new configuration file for Terragrunt, which will be autogenerated
+from an interactive setup process the first time you run `terragrunt new`.
+
+The following configuration options will be required:
```

Comment:
Where does this config live? Is it a `terragrunt.hcl` somewhere on your hard drive? Or does Terragrunt prompt you for this info?

Reply:
In the prototype, it lives in `~/.config/terragrunt/scaffold.toml`, and terragrunt prompts you for this info if the file doesn't exist.

We can also make it overridable via the command line.